raft/raft.go

```diff
@@ -397,8 +397,9 @@ func (r *raft) hardState() pb.HardState {
 	}
 }
 
-// send persists state to stable storage and then sends to its mailbox.
-func (r *raft) send(m pb.Message) {
+// sendInNextReady schedules persisting state to a stable storage and then
+// sending the messages (as part of next Ready message processing).
+func (r *raft) sendInNextReady(m pb.Message) {
```

Comment:
This is mostly just renaming a function? I actually prefer the simplicity of send(). Could we just have more detailed function description? I think raft/doc.go already explained pretty well that users of raft must consume Node.Ready(), and how it should be consumed.

Reply:
@tangcong up to you. 

I was very naturally assuming when reading raft code that r.send(...) means actually sending right now - maybe async, but without any additional actions needed to make it happen. 
This change does not have any impact for RAFT users..., just makes the 'onboarding' cost a little lower.

Reply:
Renaming the function name really breaks the simplicity. I suggest adding detailed comments to this function to help  Raft beginner better understand how it works. @ptabor

Reply:
Reverted this aspect.